main.py

```diff
@@ -1,40 +1,99 @@
-from registration import Model
-from detection import Detector
-from parse_args import parse_args_and_execute
+import argparse
+from registration import Model, register
+from detection import Detector, detect
+
+
+def parse_args_and_execute():
+    '''Parse command-line arguments and execute the appropriate function (register or detect).'''
+
+    parser = argparse.ArgumentParser(description="Registration and Detection")
+
+    subparsers = parser.add_subparsers(dest="command")
+
+    # Subcommand for registration
+    register_parser = subparsers.add_parser('register', help="Register an image")
+    register_parser.add_argument('--camera_params', type=str, required=True, help="Path to camera parameters file")
+    register_parser.add_argument('--input_image', type=str, required=True, help="Path to input image for registration")
+    register_parser.add_argument('--vol', type=str, help="Volume of the object")
+    register_parser.add_argument('--output_image', type=str, required=True, help="Path to save the registered image")
+    register_parser.add_argument('--crop_method', type=str, choices=['clicks', 'points', 'none'], required=True,
+                                 help="Method for cropping the image")
+    register_parser.add_argument('--points', type=str, nargs='*',
```

Comment:
Please, add a GUI for points selection on refernce image, as we discussed on meeting